src/gh-api-client.ts

```diff
@@ -1,11 +1,30 @@
 /* eslint-disable @typescript-eslint/naming-convention */
 import { Octokit } from "@octokit/action";
 import { retry } from "@octokit/plugin-retry";
+import { EndpointOptions, RequestInterface } from "@octokit/types";
+
+import { getSignedAuthToken } from "./inputs";
 
 export const userAgent = "GitHub multi-repository variant analysis action";
 
-export function getOctokit() {
-  return new Octokit({ userAgent, retry });
+export function getOctokitRequestInterface(): RequestInterface {
+  const octokit = new Octokit({ userAgent, retry });
+
+  const signedAuthToken = getSignedAuthToken();
+  if (signedAuthToken) {
+    return octokit.request.defaults({
```

Comment:
Can I just double check (I think that's what you said in the description, but want to make sure): Are you saying that the following doesn't work?

````ts
return octokit.request.defaults({
  headers: {
    authorization: `RemoteAuth ${signedAuthToken}`,
  },
});

```

Reply:
I think this is because the auth strategy is also a hook, so it will override any `Authorization` headers are set (see [this](https://github.com/octokit/core.js/blob/b41adec60043ed69ba1649e48822458e229fd24d/src/index.ts#L140-L159)). I think you will be able to set the header when using [`@octokit/request`](https://github.com/octokit/request.js) directly, but then you lose the defaults set by `@octokit/core` (which might include the fallback to using `GITHUB_TOKEN`).